[nigellqx] iP

nigellqx opened the pull request:
# Tako

> [!TIP]
> `"The function of good software is to make the complex appear to be simple."` - Grady Booch [source](https://quotefancy.com/quote/1663415/Grady-Booch-The-function-of-good-software-is-to-make-the-complex-appear-to-be-simple)

Tako can manage tasks that you currently have. It is 

- Text-Based
- Easy to use
- ~~Easy~~ VERY EASY to learn

Here are the steps to use it:

1. Download it
2. Add your task
3. Use different features to manage your tasks 
4. **Enjoy it!** 😃 

So here are some of the features:

- [x] _Managing task_
- [x] _Managing task with deadline_
- [x] _Managing task with a date_

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 

```Java
public class Main {
       public static void main (String[] args) {
              Application.launch(MainApp.class, args);
       }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nigellqx committed:
Rename the chatbot, implemented greeting of chatbot and also exiting of the chatbot

nigellqx committed:
Update Tako.java

The name of the chatbot is now Tako.
The name is changed to make the chatbot unique to myself

The chatbot can now greet the user and exit.
This is to allow the user to know that they enter and exit the chatbot successfully.

nigellqx committed:
Update Tako.java

The chatbot will echo the command given by the users.
If the command given is 'bye', the chatbot will exit.

nigellqx committed:
Update Tako.java

Chatbot can now store text given by the user
If the user types "list", it returns all the text stored by the chatbot

nigellqx committed:
Add Task.java
Update Tako.java

Task.java is added so that the function of marking and unmarking of task can be done in a more natural way.

nigellqx committed:
Add Deadline.java
Add Event.java
Add ToDo.java
Update Task.java
Update Tako.java

Added 3 new classes which extends from Task (inheritance)
Each task has some similarity between them but is unique in their own ways.

nigellqx committed:
Update Event.java
Update Tako.java
Update EXPECTED.TXT
Update input.txt
Update runtest.sh

Did automated testing with some inputs and fix the bugs that occur as a result of the testing

nigellqx committed:
Update Tako.java
Add InvalidCommandException

Chatbot is able to identify some invalid commands that are being passed into it

nigellqx committed:
Update Tako.java

Chatbot is now able to remove task from the list

nigellqx committed:
Update Tako.java

nigellqx committed:
Update Deadline.java
Update Tako.java

nigellqx committed:
Merge branch 'branch-Level-8'

nigellqx committed:
Update Deadline.java
Update Event.java
Update Tako.java
Update Task.java
Update TakoException.java

Add Parser.java
Add Storage.java
Add TaskList.java
Add Ui.java

nigellqx committed:
Merge branch 'branch-A-MoreOOP'

nigellqx committed:
Update TakoException.java
Update Parser.java
Update Storage.java
Update Tako.java
Update Deadline.java
Update Event.java
Update Task.java
Update TaskList.java
Update ToDo.java
Update Ui.java

nigellqx committed:
Merge branch 'branch-A-Packages'

nigellqx committed:
Merge branch 'branch-A-Packages'

nigellqx committed:
Merge branch 'add-gradle-support'

nigellqx committed:
Add build.gradle

nigellqx committed:
Merge branch 'branch-A-Gradle'

nigellqx committed:
Update Ui.java
Add ParserTest.java
Add ToDoTest.java

nigellqx committed:
Merge branch 'branch-A-JUnit'

nigellqx committed:
Update TakoException.java
Update Parser.java
Update Storage.java
Update Tako.java
Update Deadline.java
Update Event.java
Update Task.java
Update TaskList.java
Update ToDo.java
Update Ui.java

nigellqx committed:
Update TakoException.java
Update Parser.java
Update Storage.java
Update TaskList.java
Update Ui.java
Update ParserTest.java

nigellqx committed:
Update TakoException.java
Update Parser.java
Update Storage.java
Update TaskList.java
Update Ui.java

nigellqx committed:
Update Parser.java
Update Ui.java

nigellqx committed:
Merge branch 'branch-A-JavaDoc'

nigellqx committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/parser/Parser.java

nigellqx committed:
Merge branch 'branch-Level-9'

nigellqx committed:
Update Parser.java

notnotmax commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+     */
+    public static void parse(String command) {
+        try {
+            if (command.equalsIgnoreCase("bye")) {
```
I like that you thought of ignoring case when checking. Though, perhaps a switch statement could be easier to read (after preprocessing the command string)?

notnotmax commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+    /**
+     * Marks the task as complete according to the input
+     *
+     * @param command
```
Minor nitpick - this param seems to be missing a description.

notnotmax commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+     * @param command
+     * @throws TakoException if the deadline command is not in the right form
+     */
+    public static void parseDeadline(String command) {
```
The deeply nested code may be difficult to read with really specific string checking - maybe some comments would be valuable?

notnotmax commented on src/main/java/parser/Parser.java:
Overall it seems like the code works well enough, good job! However, a possible alternative to avoid finicky string matching and deeply nested code could be to make use of the split() method for strings? Is there a reason why the string checking is done using length and charAt?

nigellqx replied:
Did not think about the split() method before. Thank you for your suggestions, really helped me alot 👍

notnotmax commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    }
+
+    /**
+     * Says goodbye to the user
```
Minor nitpick - maybe add a period after the sentence for consistency with other javadoc method headers?

notnotmax commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,123 @@
+     *
+     * @param task task that is being added to the list.
+     */
+    public static void addTaskMessage(Task task) {
```
Perhaps a name like `printAddTaskMessage` would make it clearer that the method's purpose is to print something, rather than being confused with 'adding a task message'?

notnotmax commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think explicit imports would better adhere to the coding standard. Maybe this is an unintentional mistake caused by the IDE?

notnotmax commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+public class ParserTest {
+
+    @Test
+    public void isValidDateTest1() {
```
Minor nitpick - maybe some comments to describe what the test is could be useful? Alternatively, consider the naming convention `unitBeingTested_descriptionOfTestInputs_expectedOutcome` for method names to make them self-explanatory (from week 3 topics).

notnotmax reviewed (commented):
lgtm - just a few potential considerations regarding the coding quality.

nigellqx committed:
Update build.gradle
Update Parser.java
Update Storage.java

nigellqx committed:
Merge branch 'branch-A-CheckStyle'

nigellqx committed:
Update Parser.java
Update Tako.java
Update TaskList.java
Update Ui.java

Add DialogBox.java
Add Launcher.java
Add Main.java
Add MainWindow.java
Add dialog-box.css
Add main.css
Add DialogBox.fxml
Add MainWindow.fxml

nigellqx committed:
Merge branch 'branch-Level-10'

estellelim commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+    /**
+     * Helps to add the task from the file when the chatbot just starts.
+     *
+     * @param task task to be loaded into the list
```
Minor coding standard nitpick - punctuation behind each parameter description

estellelim reviewed (commented):
LGTM, do not see any other coding standard violations other than the some of the JavaDoc comments missing punctuations or description, and also the wildcard import commented on by @notnotmax
Good job overall, especially on the layout, and variable/ method naming!

nigellqx committed:
Update Parser.java

Delete command: prints the wrong output
Mark command: prints the wrong output

Bote command should output the command has been executed.

The method returns the String to be output through the method that is being called in TaskList.

This allows the Parser command to print the correct output instead of the 'Wrong Format' message.

nigellqx committed:
Update TaskList.java
Update Ui.java

TaskList class: the get function assume that the i that is passed in is positive and less than the size of task list
Ui class: the printList() function assume that the TaskList.length() function will return a positive number

Assertion is added to both of these methods to make sure the assumption holds.

This is to make sure that the code does not fail due to the carelessness of the programmer.

nigellqx committed:
Update Parser.java
Update Ui.java
Update ParserTest.java

The code have minor issues about its code quality.

The code is more readable and tidy.

The code is being changed to adhere to the coding standards.

It is done to make sure the code comply with the coding standards.

nigellqx committed:
Merge pull request #1 from nigellqx/branch-A-Assertions

Update TaskList.java

nigellqx committed:
Merge branch 'master' into branch-A-CodeQuality

nigellqx committed:
Merge pull request #2 from nigellqx/branch-A-CodeQuality

Update Parser.java

nigellqx committed:
Update Parser.java
Update Storage.java
Update Task.java
Update TaskList.java
Update Ui.java

The current implementation does not have a priority feature.

A priority feature will be useful for users to set tasks that are more urgent at the top of the list

The user can now set to HIGH, MEDIUM or LOW for each tasks with LOW being the default.

It is done this way so tasks with HIGH priority will be shown at the top.

nigellqx committed:
Merge branch 'branch-BCD-extension'

nigellqx committed:
Update Parser.java
Update Task.java
Update ParserTest.java
Update ToDoTest.java
Add TaskTest.java

nigellqx committed:
Update Storage.java
Update TaskList.java
Update Ui.java

nigellqx committed:
Update TakoException.java
Update Parser.java
Update TaskList.java

nigellqx committed:
Update Main.java
Add Ui.png

nigellqx committed:
Add Ui.png
Update README.md
Update Parser.java
Update Storage.java
Update Ui.java

nigellqx committed:
Update README.md

nigellqx committed:
Update README.md
Update TakoException.java

nigellqx committed:
Update Ui.png

nigellqx committed:
Update Ui.png
Update README.md

nigellqx committed:
Update README.md

nigellqx committed:
Update README.md
Update Ui.png
Update Tako.png
Update User.png

nigellqx committed:
Update README.md
Update MainWindow.java

nigellqx committed:
Update TakoException.java
Update Parser.java
Update Storage.java
Update Task.java
Update TaskList.java
Update Ui.java
